### PR TITLE
[vcpkg_download_distfile] Fix regression for downloading via ARIA2 and downloading with HTTP headers

### PR DESCRIPTION
**Pull request by ras0219-msft**

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Maybe fix #18291.

File: scripts/cmake/vcpkg_download_distfile.cmake [hunk 1]
```diff
@@ -162,9 +162,13 @@ function(vcpkg_download_distfile VAR)
                     ${DOWNLOADS}/download-${vcpkg_download_distfile_FILENAME}-err.log
                     ${DOWNLOADS}/download-${vcpkg_download_distfile_FILENAME}-detailed.log
                 )
+                get_filename_component(downloaded_file_dir "${downloaded_file_path}" DIRECTORY)
+                file(MAKE_DIRECTORY "${downloaded_file_dir}")
+                file(RENAME "${DOWNLOADS}/temp/${vcpkg_download_distfile_FILENAME}" "${downloaded_file_path}")
```
Comment (Neumann-A) [inline scripts/cmake/vcpkg_download_distfile.cmake:167]: missing a test_hash call here ?

`test_hash("${DOWNLOADS}/temp/${vcpkg_download_distfile_FILENAME}" "downloaded file" "The file may have been corrupted in transit.")`

**Comment by Neumann-A** [comment]
works for me. just missing a test_hash in the aria path

File: scripts/cmake/vcpkg_download_distfile.cmake (hunk 1 above)
Comment (ras0219-msft) [reply scripts/cmake/vcpkg_download_distfile.cmake:167]: Ick, thanks for the catch.

**Review by strega-nil — APPROVED** [review]

**Review by vicroms — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
All the failures are download failures, so I'm fine merging. Thanks robert!

**Pull request merged**